File: JULY-2025/05-07-2025/p1394.dart
class Solution {
  int findLucky(List<int> arr) {
    List<int> freq = [];

    int lucky = -1;
    for (int i = 0; i < arr.length; i++) {
      int current = arr[i];
      if (freq.contains(current)) {
        continue;
      }
      int count = 1;
      for (int j = i + 1; j < arr.length; j++) {
        if (arr[j] == current) {
          count++;
        }
      }
      if (count == current) {
        print('$current is a lucky integer');
        if (current > lucky) lucky = current;
      } else {
        print('$current is not a lucky integer');
      }
      freq.add(current);
    }
     return lucky;
  }
}

File: JULY-2025/05-07-2025/p1394.php
class Solution {

    /**
     * @param Integer[] $arr
     * @return Integer
     */
    function findLucky($arr) {
        $count=[];

        foreach($arr as $num)
        {
            if(isset($count[$num]))
            {
                $count[$num]++;
            }
            else{
                $count[$num]=1;
            }
        }
        $maxlucky=-1;
        foreach($count as $num=>$freq){
            if($num == $freq && $num > $maxlucky)
            {
                $maxlucky = $num;
            }
        }
        return $maxlucky;
    }
}